runtime/collections/maps.cpp

```diff
@@ -45,7 +45,7 @@ SortKItem hook_MAP_lookup_null(SortMap m, SortKItem key) {
 SortKItem hook_MAP_lookup(SortMap m, SortKItem key) {
   auto res = hook_MAP_lookup_null(m, key);
   if (!res) {
-    throw std::invalid_argument("key not found");
+    throw std::invalid_argument("Key not found for map lookup");
```

Comment:
Would it be possible to pretty print the key?
If you have a large definition with multiple places where something like this could happen, it would help narrow it down even faster.
I'm not sure if it would be feasible if the key is very large though. We might need to limit the pretty printed term size.